### PR TITLE
[ADD] feat(file-widget) Customizing the file widget section

### PR DESCRIPTION
**Pull request by Pourbaix**
Customizing the file widget to add some UI elements like access restriction and license type for each bitstream.

### CODE REVIEW

File: src/app/submission/sections/upload/file/license-type/section-upload-file-license-type.component.html [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div class="d-flex flex-row" style="max-width: fit-content;">
+    <a [href]="licenseTypeURI" target="_blank">
+        <img [src]="imageToDisplay" [style]="'max-height:' + maxHeight + 'px;'">
```
Comment (zannkukai) [inline src/app/submission/sections/upload/file/license-type/section-upload-file-license-type.component.html:3]: Maybe overkill for this feature but could be of interest to keep in mind : https://angular.io/guide/image-directive

File: src/themes/dspace/app/submission/sections/upload/file/section-upload-file.component.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    }
+}
+
+.action-btn {
+    font-size: 1.2rem;
+}
+
+.flex-grow {
+    flex-grow: 10;
+}
```
Comment (zannkukai) [inline src/themes/dspace/app/submission/sections/upload/file/section-upload-file.component.scss]: 1) no need to specify "10", only "1" should be necessary if other flexbox elements don't specify any "flex-grow".
2) you don't need to specify a class, you could use the `flex-grow-1` provided by bootstrap

File: src/themes/dspace/app/submission/sections/upload/file/section-upload-file.component.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+          <div class="d-flex flex-grow align-items-center">
+            <h3 class="m-0 text-primary">{{fileName}} <span class="text-muted">({{fileData?.sizeBytes | dsFileSize}})</span></h3>
+            <ng-container *ngIf="fileData.accessConditions && fileData.accessConditions.length">
+              <span class="ml-2" *ngIf="fileData.accessConditions[0].name == 'restricted'"><i class="fa fa-lock text-danger"></i></span>
+              <span class="ml-2" *ngIf="fileData.accessConditions[0].name == 'embargo'"><i class="fa fa-lock text-warning"></i></span>
+              <span class="ml-2" *ngIf="fileData.accessConditions[0].name == 'openaccess'"><i class="fa fa-lock-open text-success"></i></span>
+            </ng-container>
+          </div>
+          <div *ngIf="!readOnly" class="d-flex align-items-center action-list">
+            <ng-container>
```
Comment (zannkukai) [inline src/themes/dspace/app/submission/sections/upload/file/section-upload-file.component.html]: this container seems useless (no condition, no loop) --> could be removed (if it's your change)

File: src/themes/dspace/app/submission/sections/upload/file/section-upload-file.component.html [hunk 4]
```diff
@@ -0,0 +1,59 @@
+<ng-container *ngIf="fileData">
+    <div>
+      <div class="py-2">
+        <div class="d-flex align-items-center border-2 border-bottom mb-2 pb-1">
+          <div class="d-flex flex-grow align-items-center">
+            <h3 class="m-0 text-primary">{{fileName}} <span class="text-muted">({{fileData?.sizeBytes | dsFileSize}})</span></h3>
+            <ng-container *ngIf="fileData.accessConditions && fileData.accessConditions.length">
+              <span class="ml-2" *ngIf="fileData.accessConditions[0].name == 'restricted'"><i class="fa fa-lock text-danger"></i></span>
+              <span class="ml-2" *ngIf="fileData.accessConditions[0].name == 'embargo'"><i class="fa fa-lock text-warning"></i></span>
+              <span class="ml-2" *ngIf="fileData.accessConditions[0].name == 'openaccess'"><i class="fa fa-lock-open text-success"></i></span>
```
Comment (zannkukai) [inline src/themes/dspace/app/submission/sections/upload/file/section-upload-file.component.html]: `span` tags are useless :
```typescript
<i class="ml-2 fa fa-lock text-..." *ngIf="..."></i>
```

File: src/themes/dspace/app/submission/sections/upload/file/section-upload-file.component.html [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+            </ng-container>
+          </div>
+          <div *ngIf="!readOnly" class="d-flex align-items-center action-list">
```
Comment (zannkukai) [inline src/themes/dspace/app/submission/sections/upload/file/section-upload-file.component.html]: strange (for me?) to combine boostrap classes and custom class in this case.
You could only keep your "action-list" class here and define "d-flex" and "align-items-center" into your custom class.
```scss
.action-list {
    @extends d-flex;
    @extends align-items-center;
   * { height: 100%; } //seems strange too ... but if it works OK
   .fa { font-size: 1rem; }
}
```

File: src/themes/dspace/app/submission/sections/upload/section-upload.component.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
+<ds-alert [content]="'submission.sections.upload.info'"
+          [dismissible]="true"
+          [type]="AlertTypeEnum.Info"></ds-alert>
+
+<ng-container *ngIf="fileList.length == 0">
+  <div class="row">
+    <div class="col-md-12">
```
Comment (zannkukai) [inline src/themes/dspace/app/submission/sections/upload/section-upload.component.html]: no need to specify "md" if your don't specify "sm" or "lg" --> just "col-12"

File: src/themes/dspace/app/submission/sections/upload/section-upload.component.html [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+      <h3 class="text-center"><span class="text-muted">{{'submission.sections.upload.no-file-uploaded' | translate}}</span></h3>
```
Comment (zannkukai) [inline src/themes/dspace/app/submission/sections/upload/section-upload.component.html]: span seems unecessary --> `<h3 class="text-center text-muted"> ...</h3>`

File: src/themes/dspace/app/submission/sections/upload/section-upload.component.html [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    </div>
+  </div>
+</ng-container>
+
+<ng-container *ngIf="fileList.length > 0">
+
+  <div *ngIf="collectionDefaultAccessConditions.length > 0" class="row">
+    <div class="col-sm-12" >
```
Comment (zannkukai) [inline src/themes/dspace/app/submission/sections/upload/section-upload.component.html]: same here --> `col-12`

File: src/themes/dspace/app/submission/sections/upload/section-upload.component.html (hunk 7 above)
Comment (Pourbaix) [reply src/themes/dspace/app/submission/sections/upload/section-upload.component.html]: not my code :/

File: src/themes/dspace/app/submission/sections/upload/section-upload.component.html (hunk 8 above)
Comment (Pourbaix) [reply src/themes/dspace/app/submission/sections/upload/section-upload.component.html]: not my code :/

**Pull request merged**